docs/concepts.md

```diff
@@ -1,27 +1,213 @@
-## Concepts
+# Concept Reference
 
-The ksonnet framework is centered around a couple of key concepts.
+The ksonnet framework is centered around a couple of key concepts and terms. This reference gives a brief explanation of each. To get a more tangible sense of how these concepts tie together, see the Tutorial (WIP, link pending).
+
+* Ksonnet
+  * [Application](#application)
+  * [Environment](#environment)
+  * [Component](#component)
+  * [Prototype](#prototype)
+  * [Parameter](#parameter)
+  * [Part](#part)
+  * [Package](#package)
+  * [Registry](#registry)
+* Related
+  * [Manifest](#manifest)
+  * [Jsonnet](#jsonnet)
+
+
+The following diagram shows how the ksonnet framework works holistically:
+
+![ksonnet overview diagram](/docs/img/ksonnet_overview.svg)
+
+---
 
 ### Application
-TODO
+
+A ksonnet application represents a well-structured directory of Kubernetes [manifests](#manifests). This directory is autogenerated by [`ksonnet init`](docs/cli-reference/ksonnet_init.md). These manifests typically tie together in some way—collectively, they might define something like the following:
+
+<p align="center">
+<img alt="ksonnet application diagram" src="/docs/img/guestbook_app.svg" height="250px">
+</p>
+
+---
 
 ### Environment
-TODO
+
+An environment consists of four elements:
+
+| What       | Example       | Source          | How is this tracked? |
+| ---------- | ------------- | ------------- | ---------- |
+| (1) *Name* | dev | Specified at command line | A directory under `environments/` (e.g. `environments/dev/`). <br><br>**A name with slashes results in a hierarchical file structure**. For example, `us-west/staging` creates `environments/us-west/staging/`.|
+| (2) *Cluster URI* | https://cluster-name-with-hash.us-west-2.elb.amazonaws.com | Pulled from current [kubeconfig context](https://kubernetes.io/docs/concepts/configuration/organize-cluster-access-kubeconfig/#context) OR specified at command line | Inside `environments/<env-name>/spec.json` |
+| (3) *Kubernetes [namespace](https://kubernetes.io/docs/concepts/overview/working-with-objects/namespaces/)* | dev | Pulled from current [kubeconfig context](https://kubernetes.io/docs/concepts/configuration/organize-cluster-access-kubeconfig/#context) OR specified at command line | Inside `environments/<env-name>/spec.json` |
```

Comment:
I'm not sure if the `Source` column would be particularly useful. Because `namespace` and `server` can also be defined at the command line. I think the specifics of this are best left to the command line docs.

I think it might be more helpful to give a description of what each of these are.